docs/specs/build/toc/corners.yml

```diff
@@ -58,3 +51,42 @@ outputs:
           }
       ]
     }
+---
+# Circle toc referenceing 1
+inputs:
+  docfx.yml: |
+    content:
+      include:
+        - docs/**/*.md
+        - docs/**/*.yml
+  docs/toc.md: |
+    # [Topic1](index.md) #
+    ## Refernce toc
+    ### [Reference TOC 1.1](f1/toc.md)
+    ### [Reference TOC 1.2](f1/toc.md)
+    ### [Reference TOC 2.1](f2/toc.md)
+  docs/f1/toc.md: |
+    # [Topic1](../index.md)
+    ## [Reference TOC 1.1.1](f2/toc.md)
+  docs/f2/toc.md: |
+    # [Topic1](../index.md)
+    ## [Reference TOC back](../toc.md)
+  docs/index.md:
+exceptions:
+  system.applicationexception: |
+    Circle toc reference was detected, docs/toc.md-->docs/f2/toc.md-->docs/toc.md
+---
+# Circle toc referenceing 2
+inputs:
+  docfx.yml: |
+    content:
+      include:
+        - docs/**/*.md
+        - docs/**/*.yml
+  docs/toc.md: |
+    # [Topic1](index.md) #
+    ## [Reference itself](toc.md)
+  docs/index.md:
+exceptions:
+  system.applicationexception: |
+    Circle toc reference was detected, docs/toc.md-->docs/toc.md
```

Comment:
Add a case like this, there should be no circular reference:

`a` -> `b` -> `c`
`a` -> `c`

Reply:
already have it

---
In reply to: [185987606](https://github.com/dotnet/docfx/pull/2725#discussion_r185987606) [](ancestors = 185987606)